doc/fuzzing.md

```diff
@@ -183,30 +181,26 @@ There are 3 ways fuzz tests can be built:
    tests would not be useful. This build is only useful for ensuring fuzz tests
    compile and link.
 
-## macOS hints for libFuzzer
+## macOS notes
 
-The default Clang/LLVM version supplied by Apple on macOS does not include
-fuzzing libraries, so macOS users will need to install a full version, for
-example using `brew install llvm`.
+Support for fuzzing on macOS is not officially maintained by this project. If
+you are running into issues on macOS, we recommend fuzzing on Linux instead for
+best results. On macOS this can be done within Docker or a virtual machine.
 
-You may also need to take care of giving the correct path for `clang` and
-`clang++`, like `CC=/path/to/clang CXX=/path/to/clang++` if the non-systems
-`clang` does not come first in your path.
+Reproducing and debugging fuzz testcases on macOS is supported, by building the
+fuzz binary without support for any specific fuzzing engine.
 
-Using `lld` is required due to issues with Apple's `ld` and `LLVM`.
+You may still be able to fuzz on macOS using the following steps (provided on
+best effort basis, only aimed at the latest macOS version, may be outdated):
 
-Full configuration step for macOS:
-
-```sh
+```
 $ brew install llvm lld
-$ cmake --preset=libfuzzer \
+$ cmake --preset=libfuzzer-nosan \
    -DCMAKE_C_COMPILER="$(brew --prefix llvm)/bin/clang" \
    -DCMAKE_CXX_COMPILER="$(brew --prefix llvm)/bin/clang++" \
    -DCMAKE_EXE_LINKER_FLAGS="-fuse-ld=lld"
```

Comment:
since this is an alternative to the above linux one and since the build directory is changed now, should we add
```md
cmake --build build_fuzz_nosan -j$(nproc)
FUZZ=process_message build_fuzz_nosan/bin/fuzz
```
after this?